src/utils.c

```diff
@@ -123,19 +126,23 @@ SEXP colnamesInt(SEXP x, SEXP cols, SEXP check_dups) {
       ricols = PROTECT(coerceVector(cols, INTSXP)); protecti++;
     }
     int *icols = INTEGER(ricols);
-    for (int i=0; i<nc; i++) {
-      if ((icols[i]>nx) || (icols[i]<1))
-        error(_("argument specifying columns received non-existing column(s): cols[%d]=%d"), i+1, icols[i]); // handles NAs also
+    if(!Skip_absent){
```

Comment:
for this case, I think we need more nuanced behavior.

`skip_absent` is meant to flexibly allow user to say `cols=8`, but perhaps the table only has 7 columns, then nothing happens (instead of being an error).

But I think `cols<0` still makes no sense -- that's an invalid number of columns. So I'd suggest still erroring if `icols[i] < 0`.

That leaves the case of `cols==0`... that's a valid _number_ of columns, but it's never a valid _column number_. So I think `icols[i] == 0` should also be an error, do you agree?

Reply:
yeah i think so, should i set a different type of error  like "invalid number of column" or just allow this error to get triggered when these condition happen

Reply:
just leave the error as is for now, eventually #6008 can serve to improve the message quality

Reply:
I was thinking of implementing something like this :
```
 for (int i=0; i<nc; ++i) {
      if ((icols[i]>nx && !bskip_absent) || (icols[i]<1))
        error(_("argument specifying columns received non-existing column(s): cols[%d]=%d"), i+1, icols[i]); // handles NAs also
    }
 ```
 but yeah this gives error when `Na` is used as well, what i wanna know is , do we want `Na` to work or not

Reply:
that looks roughly right to me. let's defer on the issue of NA til a future PR.

Reply:
ok, then i am changing the tests containing the Na as well 👍